ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/API.hs

```diff
@@ -423,6 +426,10 @@ data AddBlockPromise m blk = AddBlockPromise
       -- disk.
     }
 
+data Processed a = Processed a
```

Comment:
I think this naming doesn't provide more value than `Either String a`. Would making it more specific be a good thing? like

```haskell
data AddBlockResult blk = SuccesfullyAddedBlock (Point blk)
                        | FailedToAddBlock String
```

Do we need to use it with something that is not a `Point blk`?

Reply:
Yes I like this :) Thank you for the suggestion!